File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # for i in nums:
        #     #print(i)
        #     x = abs(target-i)
        #     #print(x)
        #     if x in nums:
        #         z = nums.index(i)
        #         break
        # for j in range(0,len(nums)):
        #     if(nums[j] == x):
        #         y = j
        #         return (z,y)
        #     # if x in nums:
        #     #     return (nums.index(i),nums.index(x))
        for i in range(0,len(nums)):
            for j in range(i+1,len(nums)):
                if(nums[i] + nums[j] == target):
                    return (i,j)

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {

        Map<Integer,Integer> mp = new HashMap<>();
        int n = nums.length;
        for(int i = 0 ; i < n ; i++) 
        {
            int c = target - nums[i];
            if(mp.containsKey(c))
            {
                return new int[]{mp.get(c),i};
            }
            mp.put(nums[i],i);

        }
        return new int[]{};
    }
}